python/lsst/ts/ess/labjack/base_labjack_data_client.py

```diff
@@ -0,0 +1,177 @@
+from typing import Any
+
+# Hide mypy error `Module "labjack" has no attribute "ljm"`
+from labjack import ljm  # type: ignore
+
+from lsst.ts.ess import common
+
+from lsst.ts import salobj
+
+# Time limit for connecting to the LabJack (seconds)
+CONNECT_TIMEOUT = 5
+
+# LabJack's special identifier to run in simulation mode.
+MOCK_IDENTIFIER = "LJM_DEMO_MODE"
+
+
+class BaseLabJackDataClient(common.BaseDataClient):
+    """Base class for ESS data clients that read a LabJack T7 or similar.
+
+    Parameters
+    ----------
+    name : str
+    config : types.SimpleNamespace
+        The configuration, after validation by the schema returned
+        by `get_config_schema` and conversion to a types.SimpleNamespace.
+    topics : `salobj.Controller`
+        The telemetry topics this model can write, as a struct with attributes
+        such as ``tel_temperature``.
+    log : `logging.Logger`
+        Logger.
+    simulation_mode : `int`, optional
+        Simulation mode; 0 for normal operation.
+
+    Notes
+    -----
+    In simulation mode the mock LabJack returns unspecified values,
+    and those values may change in future versions of the LabJack software.
+    """
+
+    def __init__(
+        self,
+        config: types.SimpleNamespace,
+        topics: salobj.Controller | types.SimpleNamespace,
+        log: logging.Logger,
+        simulation_mode: int = 0,
+    ) -> None:
+        # handle to LabJack device
+        self.handle = None
+
+        # The thread pool executor used by `run_in_thread`.
+        self._thread_pool = concurrent.futures.ThreadPoolExecutor(max_workers=1)
+
+        super().__init__(
+            config=config, topics=topics, log=log, simulation_mode=simulation_mode
+        )
+
+    @property
+    def connected(self) -> bool:
+        return self.handle is not None
+
+    def descr(self) -> str:
+        return f"identifier={self.config.identifier}"
+
+    async def run_in_thread(self, func: Callable[[], Any], timeout: float) -> Any:
+        """Run a blocking function in a thread pool executor.
+
+        Only one function will run at a time, because all calls use the same
+        thread pool executor, which only has a single thread.
+
+        Parameters
+        ----------
+        func : `Callable`
+            The blocking function to run;
+            The function must take no arguments.
+            For example: ``self._blocking_connect``.
+        timeout : `float`
+            Time limit (seconds).
+        """
+        loop = asyncio.get_running_loop()
+        try:
+            return await asyncio.wait_for(
+                loop.run_in_executor(self._thread_pool, func), timeout=timeout
+            )
+        except Exception:
+            self.log.exception(f"Blocking function {func} failed")
+            raise
+
+    async def connect(self) -> None:
+        """Connect to the LabJack.
+
+        Disconnect first, if connected.
+
+        Raises
+        ------
+        RuntimeError
```

Comment:
Please use back ticks around types.

Reply:
Are they wanted here? I've never done that before in Raises sections.